### PR TITLE
libjack2: fix cross compilation

### PR DESCRIPTION
**Pull request by mikatammi**
###### Description of changes

Fix cross compilation, by substituting the svnversion_regenerate.sh script's bash with buildPackages.bash, instead of host's bash.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-linux.pkgsCross.aarch64-multiplatform
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mikatammi** [comment]
I was a bit unsure is it good style to use `buildPackages` this way, or are there better ways?

File: pkgs/misc/jackaudio/default.nix [hunk 1]
```diff
@@ -48,7 +49,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   prePatch = ''
     substituteInPlace svnversion_regenerate.sh \
-        --replace /bin/bash ${bash}/bin/bash
+        --replace /bin/bash ${buildPackages.bash}/bin/bash
```
Comment (Artturin) [inline pkgs/misc/jackaudio/default.nix:52]: If it's a shebang then
`patchShebangs --build svnversion_regenerate.sh`

Comment (mikatammi) [reply pkgs/misc/jackaudio/default.nix]: Done

**Comment by Artturin** [comment]
Has to target staging https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

**Comment by mikatammi** [comment]
> Has to target staging https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging

Retargeted to staging now

**Review by NickCao — APPROVED** [review]

**Comment by NickCao** [comment]
LGTM, one nitpick: prePatch -> postPatch

**Review by Artturin — APPROVED** [review]

**Comment by mikatammi** [comment]
> LGTM, one nitpick: prePatch -> postPatch

Fixed now!

**Comment by mikatammi** [comment]
Apparently the broken libjack2 has now entered NixOS-23.05 :)

Could we backport this when this gets from staging to master?

**Pull request merged**